Backup Feature making Errors

[00:48:48 WARN]: java.lang.reflect.InvocationTargetException
[00:48:48 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
[00:48:48 WARN]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.execute(DefaultHandler.java:79)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.lambda$handle$0(DefaultHandler.java:67)
[00:48:48 WARN]: 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
[00:48:48 WARN]: 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[00:48:48 WARN]: 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
[00:48:48 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[00:48:48 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[00:48:48 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1589)
[00:48:48 WARN]: Caused by: java.io.IOException: The process cannot access the file because another process has locked a portion of the file
[00:48:48 WARN]: 	at java.base/sun.nio.ch.FileDispatcherImpl.read0(Native Method)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.FileDispatcherImpl.read(FileDispatcherImpl.java:54)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:339)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:305)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:282)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:240)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:74)
[00:48:48 WARN]: 	at java.base/sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:103)
[00:48:48 WARN]: 	at java.base/java.io.InputStream.read(InputStream.java:217)
[00:48:48 WARN]: 	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1309)
[00:48:48 WARN]: 	at org.apache.commons.io.IOUtils.copy(IOUtils.java:978)
[00:48:48 WARN]: 	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1282)
[00:48:48 WARN]: 	at org.apache.commons.io.IOUtils.copy(IOUtils.java:953)
[00:48:48 WARN]: 	at org.apache.commons.io.IOUtils.toByteArray(IOUtils.java:2405)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.controller.BackupController.zipDirectory(BackupController.java:54)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.controller.BackupController.zipDirectory(BackupController.java:51)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.controller.BackupController.zipDirectory(BackupController.java:51)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.controller.BackupController.createBackup(BackupController.java:85)
[00:48:48 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.panel.routes.backups.BackupRoute.put(BackupRoute.java:98)
[00:48:48 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[00:48:48 WARN]: 	... 9 more
[01:00:35 WARN]: java.lang.reflect.InvocationTargetException
[01:00:35 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
[01:00:35 WARN]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.execute(DefaultHandler.java:79)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.lambda$handle$0(DefaultHandler.java:67)
[01:00:35 WARN]: 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
[01:00:35 WARN]: 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[01:00:35 WARN]: 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
[01:00:35 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[01:00:35 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[01:00:35 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1589)
[01:00:35 WARN]: Caused by: java.lang.OutOfMemoryError: Java heap space
[01:00:35 WARN]: 	at java.base/java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:71)
[01:00:35 WARN]: 	at java.base/java.nio.ByteBuffer.allocate(ByteBuffer.java:373)
[01:00:35 WARN]: 	at jdk.httpserver/sun.net.httpserver.Request$WriteStream.write(Request.java:387)
[01:00:35 WARN]: 	at jdk.httpserver/sun.net.httpserver.FixedLengthOutputStream.write(FixedLengthOutputStream.java:81)
[01:00:35 WARN]: 	at java.base/java.io.FilterOutputStream.write(FilterOutputStream.java:108)
[01:00:35 WARN]: 	at jdk.httpserver/sun.net.httpserver.PlaceholderOutputStream.write(ExchangeImpl.java:459)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.http.ResponseController.send(ResponseController.java:158)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.http.ResponseController.bytes(ResponseController.java:87)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.panel.routes.backups.BackupDownloadRoute.get(BackupDownloadRoute.java:38)
[01:00:35 WARN]: 	at java.base/java.lang.invoke.LambdaForm$DMH/0x000000080268c000.invokeVirtual(LambdaForm$DMH)
[01:00:35 WARN]: 	at java.base/java.lang.invoke.LambdaForm$MH/0x00000008010b4400.invoke(LambdaForm$MH)
[01:00:35 WARN]: 	at java.base/java.lang.invoke.LambdaForm$MH/0x00000008010d0800.invokeExact_MT(LambdaForm$MH)
[01:00:35 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:156)
[01:00:35 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[01:00:35 WARN]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.execute(DefaultHandler.java:79)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.lambda$handle$0(DefaultHandler.java:67)
[01:00:35 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler$$Lambda$8948/0x0000000802c1c870.run(Unknown Source)
[01:00:35 WARN]: 	... 6 more
[01:01:05 WARN]: java.lang.reflect.InvocationTargetException
[01:01:05 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
[01:01:05 WARN]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[01:01:05 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.execute(DefaultHandler.java:79)
[01:01:05 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.lambda$handle$0(DefaultHandler.java:67)
[01:01:05 WARN]: 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
[01:01:05 WARN]: 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[01:01:05 WARN]: 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
[01:01:05 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[01:01:05 WARN]: 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[01:01:05 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1589)
[01:01:05 WARN]: Caused by: java.lang.OutOfMemoryError: Java heap space
[01:01:05 WARN]: 	at java.base/java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:71)
[01:01:05 WARN]: 	at java.base/java.nio.ByteBuffer.allocate(ByteBuffer.java:373)
[01:01:05 WARN]: 	at jdk.httpserver/sun.net.httpserver.Request$WriteStream.write(Request.java:387)
[01:01:05 WARN]: 	at jdk.httpserver/sun.net.httpserver.FixedLengthOutputStream.write(FixedLengthOutputStream.java:81)
[01:01:05 WARN]: 	at java.base/java.io.FilterOutputStream.write(FilterOutputStream.java:108)
[01:01:05 WARN]: 	at jdk.httpserver/sun.net.httpserver.PlaceholderOutputStream.write(ExchangeImpl.java:459)
[01:01:05 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.http.ResponseController.send(ResponseController.java:158)
[01:01:05 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.http.ResponseController.bytes(ResponseController.java:87)
[01:01:05 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.panel.routes.backups.BackupDownloadRoute.get(BackupDownloadRoute.java:38)
[01:01:05 WARN]: 	at java.base/java.lang.invoke.LambdaForm$DMH/0x000000080268c000.invokeVirtual(LambdaForm$DMH)
[01:01:05 WARN]: 	at java.base/java.lang.invoke.LambdaForm$MH/0x00000008034f4400.invoke(LambdaForm$MH)
[01:01:05 WARN]: 	at java.base/java.lang.invoke.LambdaForm$MH/0x00000008010d0800.invokeExact_MT(LambdaForm$MH)
[01:01:06 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:156)
[01:01:06 WARN]: 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[01:01:06 WARN]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[01:01:06 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.execute(DefaultHandler.java:79)
[01:01:06 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler.lambda$handle$0(DefaultHandler.java:67)
[01:01:06 WARN]: 	at MCDash-1.1.4.jar//de.gnmyt.mcdash.api.handler.DefaultHandler$$Lambda$8948/0x0000000802c1c870.run(Unknown Source)
[01:01:06 WARN]: 	... 6 more

--- a/src/main/java/de/gnmyt/mcdash/api/controller/BackupController.java
+++ b/src/main/java/de/gnmyt/mcdash/api/controller/BackupController.java
@@ -42,17 +42,18 @@ public class BackupController {
      * @param directory The directory to zip
      * @param parent The parent of the directory
      * @param zipOutputStream The zip output stream
-     * @throws IOException An exception that will be thrown if the directory could not be zipped
      */
-    private void zipDirectory(File directory, String parent, ZipOutputStream zipOutputStream) throws IOException {
+    private void zipDirectory(File directory, String parent, ZipOutputStream zipOutputStream) {
         if (directory.getName().equals(backupFolder.getName())) return;
         for (File file : Objects.requireNonNull(directory.listFiles())) {
             if (file.isDirectory()) {
                 zipDirectory(file, parent + "/" + file.getName(), zipOutputStream);
             } else {
-                zipOutputStream.putNextEntry(new ZipEntry(parent + "/" + file.getName()));
-                zipOutputStream.write(IOUtils.toByteArray(Files.newInputStream(file.toPath())));
-                zipOutputStream.closeEntry();
+                try {
+                    zipOutputStream.putNextEntry(new ZipEntry(parent + "/" + file.getName()));
+                    zipOutputStream.write(IOUtils.toByteArray(Files.newInputStream(file.toPath())));
+                    zipOutputStream.closeEntry();
+                } catch (Exception ignored) { }
             }
         }
     }
@@ -61,12 +62,13 @@ public class BackupController {
      * Writes a file to a zip output stream
      * @param file The file to write
      * @param zipOutputStream The zip output stream
-     * @throws IOException An exception that will be thrown if the file could not be written
      */
-    private void zipFile(File file, ZipOutputStream zipOutputStream) throws IOException {
-        zipOutputStream.putNextEntry(new ZipEntry(file.getName()));
-        zipOutputStream.write(IOUtils.toByteArray(Files.newInputStream(file.toPath())));
-        zipOutputStream.closeEntry();
+    private void zipFile(File file, ZipOutputStream zipOutputStream) {
+        try {
+            zipOutputStream.putNextEntry(new ZipEntry(file.getName()));
+            zipOutputStream.write(IOUtils.toByteArray(Files.newInputStream(file.toPath())));
+            zipOutputStream.closeEntry();
+        } catch (Exception ignored) { }
     }
 
     /**

--- a/src/main/java/de/gnmyt/mcdash/panel/routes/backups/BackupRoute.java
+++ b/src/main/java/de/gnmyt/mcdash/panel/routes/backups/BackupRoute.java
@@ -78,7 +78,8 @@ public class BackupRoute extends DefaultHandler {
             }
 
             if (backupMode == BackupMode.SERVER) {
-                directories.add(new File("."));
+                File[] serverFolder = new File(".").listFiles();
+                directories.addAll(Arrays.asList(serverFolder != null ? serverFolder : new File[0]));
                 break;
             }
 

--- a/webui/src/states/Root/pages/Backups/components/BackupCreationDialog/BackupCreationDialog.jsx
+++ b/webui/src/states/Root/pages/Backups/components/BackupCreationDialog/BackupCreationDialog.jsx
@@ -12,7 +12,7 @@ import {
     Stack
 } from "@mui/material";
 import React, {useContext, useEffect, useState} from "react";
-import {putRequest} from "@/common/utils/RequestUtil.js";
+import {request} from "@/common/utils/RequestUtil.js";
 import {BackupContext} from "@/states/Root/pages/Backups/contexts/Backups";
 
 export const BackupCreationDialog = ({open, setOpen, setLoading}) => {
@@ -34,7 +34,7 @@ export const BackupCreationDialog = ({open, setOpen, setLoading}) => {
     const executeAction = () => {
         setOpen(false);
         setLoading(true);
-        putRequest("backups/", {mode: modes.join("")}).then(() => {
+        request("backups/", "PUT", {mode: modes.join("")}, {}, false).then(() => {
             setFinished(true);
             setLoading(false);
             updateBackups();

